### PR TITLE
Moving SQL Server and SQL Instance AAD from ActiveDirectoryClient to MicrosoftGraphClient

### PR DESCRIPTION
**Pull request by jard285**
This PR moves authentication for AzureSqlServerActiveDirectoryAdministratorAdapter and AzureSqlInstanceActiveDirectoryAdministratorAdapter from using the ActiveDirectoryClient to the new MicrosoftGraphClient

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by isra-fel** [comment]
Hey @jaredwelsh if you are Microsoft employee please visit https://aka.ms/AzureGitHub and join Azure orgnization. Thanks

File: src/Sql/Sql/ServerActiveDirectoryAdministrator/Services/AzureSqlServerActiveDirectoryAdministratorAdapter.cs [hunk 1]
```diff
@@ -40,30 +42,24 @@ public class AzureSqlServerActiveDirectoryAdministratorAdapter
...
-                if (_activeDirectoryClient == null)
+                if (_microsoftGraphClient == null)
                 {
-                    _activeDirectoryClient = new ActiveDirectoryClient(Context);
-                    if (!Context.Environment.IsEndpointSet(AzureEnvironment.Endpoint.Graph))
-                    {
-                        throw new ArgumentException(string.Format(Microsoft.Azure.Commands.Sql.Properties.Resources.InvalidGraphEndpoint));
-                    }
-                    _activeDirectoryClient = new ActiveDirectoryClient(Context);
+                    _microsoftGraphClient = AzureSession.Instance.ClientFactory.CreateArmClient<MicrosoftGraphClient>(Context, AzureEnvironment.Endpoint.Graph);
```
Comment (isra-fel) [inline src/Sql/Sql/ServerActiveDirectoryAdministrator/Services/AzureSqlServerActiveDirectoryAdministratorAdapter.cs:58]: Please set the tenant ID for the client:

`_microsoftGraphClient.TenantId = Context.Tenant.Id;` (need validation)

**Review by isra-fel — CHANGES_REQUESTED** [review]

**Comment by isra-fel** [comment]
Another thing is, we want to inform the user we have migrated to MSGraphClient. Also mentioning how to troubleshoot issues related to MSGraph permissions. So please:

1. Copy this file [MSGraphMessageHelper.cs](https://github.com/Azure/azure-powershell/blob/main/src/KeyVault/KeyVault/Helpers/MSGraphMessageHelper.cs) to your solution (update the namespace).
2. Call `WriteMessageForCmdletsSwallowException()` or `WriteMessageForCmdletsSwallowException()` (call the right cmdlet according to whether your cmdlet throws or swallows exceptions thrown from MSGraphClient) at the beginning of the cmdlets that are affected by the migration.

Thanks!

**Comment by jaredwelsh** [comment]
> Another thing is, we want to inform the user we have migrated to MSGraphClient. Also mentioning how to troubleshoot issues related to MSGraph permissions. So please:
> 
> 1. Copy this file [MSGraphMessageHelper.cs](https://github.com/Azure/azure-powershell/blob/main/src/KeyVault/KeyVault/Helpers/MSGraphMessageHelper.cs) to your solution (update the namespace).
> 2. Call `WriteMessageForCmdletsSwallowException()` or `WriteMessageForCmdletsSwallowException()` (call the right cmdlet according to whether your cmdlet throws or swallows exceptions thrown from MSGraphClient) at the beginning of the cmdlets that are affected by the migration.
> 
> Thanks!

hey Isra, for number 2, you posted the same function twice. Are there two different functions to call?

**Comment by isra-fel** [comment]
```csharp
        internal static void WriteMessageForCmdletsSwallowException(Cmdlet cmdlet);
        internal static void WriteMessageForCmdletsThrowException(Cmdlet cmdlet);
```
These two, depending on how your code handles exception from the graph client - whether it is swallowed or thrown.

**Review by isra-fel — APPROVED** [review]

**Pull request merged**